I was just compiling this locally and my compiler is complaining about some of the types in std::min:

```
NativeProcessProtocol.cpp:679:31: error: no matching function for call to 'min'
  679 |       const size_t to_write = std::min(size, sbp_addr - addr);
/include/c++/v1/__algorithm/min.h:35:1: note: candidate template ignored: deduced conflicting types for parameter '_Tp' ('size_t' (aka 'unsigned long') vs. 'unsigned long long')
   35 | min(_LIBCPP_LIFETIMEBOUND const _Tp& __a, _LIBCPP_LIFETIMEBOUND const _Tp& __b) {
```

--- a/lldb/source/Host/common/NativeProcessProtocol.cpp
+++ b/lldb/source/Host/common/NativeProcessProtocol.cpp
@@ -676,7 +676,7 @@ Status NativeProcessProtocol::WriteMemory(lldb::addr_t addr, const void *buf,
     // If the address is before a breakpoint site, write up to the site, or to
     // the end of the write. Whichever comes first.
     if (addr < sbp_addr) {
-      const size_t to_write = std::min(size, sbp_addr - addr);
+      const size_t to_write = std::min(size, size_t{sbp_addr - addr});
       size_t part_bytes_written = 0;
       error = DoWriteMemory(addr, byte_buf, to_write, part_bytes_written);
       bytes_written += part_bytes_written;

--- a/lldb/test/API/functionalities/breakpoint/write_over_software_breakpoint/main.c
+++ b/lldb/test/API/functionalities/breakpoint/write_over_software_breakpoint/main.c
@@ -17,9 +17,6 @@ int foo() {
   // in the very first instruction of foo and hoping whatever comes before is
   // not important.
   asm volatile(".globl place_break_here\n"
-#ifdef _WIN32
-               ".def place_break_here; .scl 2; .type 32; .endef;\n"
-#endif
                "place_break_here:\n"
                // The test will repeatedly add and remove a breakpoint here.
                "nop");

